src/components/headerSearch/headerSearch.tsx

```diff
@@ -0,0 +1,117 @@
+import React, { useEffect, useState } from "react";
+import Mark from "mark.js";
+import './headerSearch.css';
+import { IonButton, IonButtons, IonIcon, IonInput, IonItemDivider } from "@ionic/react";
+import { arrowDown, arrowUp } from 'ionicons/icons';
+
+interface Props {
+  doc: any
+}
+
+const HeaderSearch: React.FC<Props> = ({doc}) => {
+  const [needle, setNeedle] = useState<string>('');
+  const [marks, setMarks] = useState<NodeListOf<HTMLElement>>(document.querySelectorAll('mark'));
+  const [currentIndex, setCurrentIndex] =  useState<number>(0);
+  let mark: any;
+
+  useEffect(() => {
+    search()
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [needle]); 
+
+  const jumpTo = () => {
+    if (marks.length > 0) {
+      const currentMark = marks[currentIndex];
+      marks.forEach(mark => mark.classList.remove('current'));
+      if (currentMark) {
+        currentMark.classList.add('current');
+        currentMark.scrollIntoView({
+          behavior: "smooth",
+          inline: "start"
+        });
+      }
+    }
+  }
+
+  const jumpThrough = () => {
+    if (marks.length > 0) {
+      if (currentIndex < 0) {
+        setCurrentIndex(marks.length - 1);
+      }else if (currentIndex > marks.length - 1) {
+        setCurrentIndex(0);
+      }
+      jumpTo();
+    }
+  }
+  const jumpToNext = () => {
+    setCurrentIndex(currentIndex + 1);
```

Comment:
Very minor comment (up to you): this means that your state temporarily becomes invalid (eg. -1 or greater than the length of the array). You should always ensure that the value of your state is valid, so that the rest of your code can assume that. Perhaps do something like this:

```js
function updateCurrentIndex(ix) {
  if (ix < 0) ix = marks.length - 1;
  if (ix > marks.length - 1) ix = 0;
  setCurrentIndex(ix);
}
```

then call that from `jumpToNext` and `jumpToPrevious`,  rather than `setCurrentIndex` directly. That way your internal state is always valid, and jumpThrough doesn't need to exist.